mlflow/entities/trace_data.py

```diff
@@ -0,0 +1,22 @@
+from dataclasses import asdict, dataclass, field
+from typing import Any, Dict, List
+
+from mlflow.entities.span import Span
+
+
+@dataclass
+class TraceData:
+    """A container object that holds the spans data of a trace.
+
+    Args:
+        spans: List of spans that are part of the trace.
+    """
+
+    spans: List[Span] = field(default_factory=list)
+
+    @classmethod
+    def from_dict(cls, d):
+        return cls(spans=[Span(**span) for span in d["spans"]])
```

Comment:
```suggestion
        return cls(spans=[Span.from_dict(**span) for span in d["spans"]])
```

We might need to use `from_dict` here and cast fields.